Modules/_io/iobase.c

```diff
@@ -572,26 +572,20 @@ _io__IOBase_readline_impl(PyObject *self, Py_ssize_t limit)
                 Py_DECREF(readahead);
                 goto fail;
             }
-            if (PyBytes_GET_SIZE(readahead) > 0) {
-                Py_ssize_t n = 0;
-                const char *buf = PyBytes_AS_STRING(readahead);
-                if (limit >= 0) {
+            {
+                const Py_ssize_t len_readahead = PyBytes_GET_SIZE(readahead);
+                if (len_readahead > 0) {
+                    const Py_ssize_t upper_limit = (((limit >= 0) ? Py_MIN(limit, len_readahead) : len_readahead)-1);
+                    Py_ssize_t n = -1;
```

Comment:
This initialization is a bit awkward. Can we "add 1" to everything? i.e., make this `n = 0` and remove the `-1` above?

Reply:
Well, I did it that way so we can use prefix-++ instead of suffix-++ because the former is believed to be faster. What do you think?

Reply:
I doubt it makes any difference with a modern compiler.